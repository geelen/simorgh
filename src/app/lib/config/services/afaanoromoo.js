import { C_POSTBOX } from '@bbc/psammead-styles/colours';
import { latin } from '@bbc/gel-foundations/scripts';
import { afaanoromoo as brandSVG } from '@bbc/psammead-assets/svgs';
import '@bbc/moment-timezone-include/tz/Africa/Addis_Ababa';

const service = {
  lang: `om-ET`,
  articleAuthor: `https://www.facebook.com/bbcnewsafaanoromoo`,
  articleTimestampPrefix: 'Updated',
  atiAnalyticsAppName: 'news-afaanoromoo',
  atiAnalyticsProducerId: '2',
  brandName: 'BBC News Afaan Oromoo',
  product: 'BBC News Afaan Oromoo',
  defaultImage: 'https://news.files.bbci.co.uk/ws/img/logos/og/afaanoromoo.png',
  defaultImageAltText: 'BBC News Afaan Oromoo',
  dir: `ltr`,
  externalLinkText: ', external',
  imageCaptionOffscreenText: 'Image caption, ',
  videoCaptionOffscreenText: 'Video caption, ',
  audioCaptionOffscreenText: 'Audio caption',
  defaultCaptionOffscreenText: 'Caption, ',
  imageCopyrightOffscreenText: 'Image source, ',
  locale: `om-ET`,
  datetimeLocale: `om-ET`.toLowerCase(),
  service: 'afaanoromoo',
  serviceName: 'News Afaan Oromoo',
  themeColor: `${C_POSTBOX}`,
  twitterCreator: '@BBCAfrica',
  twitterSite: '@BBCAfrica',
  noBylinesPolicy: 'https://www.bbc.com/news/help-41670342#authorexpertise',
  publishingPrinciples: 'https://www.bbc.com/news/help-41670342',
  script: latin,
  manifestPath: '/manifest.json',
  swPath: '/sw.js',
  translations: {
    error: {
      404: {
        statusCode: '404',
        title: '404 - Fuulli kun hin argamne',
        message:
          'Tarii kun kan uumame sababa maqaa marsariitii doggoggorteefi. Maaloo maqichaa fi qubee sirreeffadhu',
        solutions: [
          'Double checking the url',
          'Hitting the refresh button in your browser',
          'Searching for this page using the BBC search bar',
        ],
        callToActionFirst: 'Alternatively, please visit the ',
        callToActionLinkText: 'BBC Afaan Oromoo Fuula duraa',
        callToActionLast: '',
        callToActionLinkUrl: 'https://www.bbc.com/afaanoromoo',
      },
      500: {
        statusCode: '500',
        title: '500 - Dogogorawwaan',
        message: 'Dogoggorri uumameera. Maaloo fuulicha haaressi',
        solutions: [
          'Hitting the refresh button in your browser',
          'Coming back again later',
        ],
        callToActionFirst: 'Alternatively, please visit the ',
        callToActionLinkText: 'BBC Afaan Oromoo Fuula duraa',
        callToActionLast: '',
        callToActionLinkUrl: 'https://www.bbc.com/afaanoromoo',
      },
    },
    consentBanner: {
      privacy: {
        title: "We've updated our Privacy and Cookies Policy",
        description: {
          uk: {
            first:
              "We've made some important changes to our Privacy and Cookies Policy and we want you to know what this means for you and your data.",
            linkText: null,
            last: null,
            linkUrl: null,
          },
          international: {
            first:
              "We've made some important changes to our Privacy and Cookies Policy and we want you to know what this means for you and your data.",
            linkText: null,
            last: null,
            linkUrl: null,
          },
        },
        accept: 'OK',
        reject: "Find out what's changed",
        rejectUrl: 'https://www.bbc.co.uk/usingthebbc/your-data-matters',
      },
      cookie: {
        title: 'Let us know you agree to cookies',
        description: {
          uk: {
            first: 'We use ',
            linkText: 'cookies',
            last:
              ' to give you the best online experience. Please let us know if you agree to all of these cookies.',
            linkUrl:
              'https://www.bbc.co.uk/usingthebbc/cookies/what-do-i-need-to-know-about-cookies/',
          },
          international: {
            first: 'We and our partners use technologies, such as ',
            linkText: 'cookies',
            last:
              ', and collect browsing data to give you the best online experience and to personalise the content and advertising shown to you. Please let us know if you agree.',
            linkUrl:
              'https://www.bbc.co.uk/usingthebbc/cookies/what-do-i-need-to-know-about-cookies/',
          },
        },
        accept: 'Yes, I agree',
        reject: 'No, take me to settings',
        rejectUrl:
          'https://www.bbc.co.uk/usingthebbc/cookies/how-can-i-change-my-bbc-cookie-settings/',
      },
    },
    media: {
      audio: 'Sagalee',
      photogallery: 'Kuusaa Fakkii',
      video: 'Viidiyoo',
    },
  },
  brandSVG,
  footer: {
    externalLink: {
      href: 'https://www.bbc.co.uk/help/web/links/',
      text: 'Itti dhiyaachuu keenya gara geessituu.',
    },
    links: [
      {
        href: 'https://www.bbc.com/afaanoromoo/institutional-49281861',
        text: 'BBC News maaliif amanuu dandeessa',
      },
      {
        href: 'https://www.bbc.co.uk/usingthebbc/terms/',
        text: 'Haala itti fayyadamaa',
      },
      {
        href: 'https://www.bbc.co.uk/usingthebbc/privacy/',
        text: 'Imaammata mateenyaa',
      },
      {
        href: 'https://www.bbc.com/usingthebbc/cookies/',
        text: 'Kuus-yaadannoo',
      },
      {
        href: 'https://www.bbc.com/afaanoromoo/institutional-42228538',
        text: 'BBC qunnami',
      },
    ],
    copyrightText: "BBC. Qabiyyee iddoola alaatiif BBC'n itti hin gaafatamu.",
  },
  fonts: [],
  navigation: [
    {
      title: 'Oduu',
      url: '/afaanoromoo',
    },
    {
      title: 'Itoophiyaa',
      url: '/afaanoromoo/topics/e986aff5-6b26-4638-b468-371d1d9617b4',
    },
    {
      title: 'Viidiyoo',
      url: '/afaanoromoo/media/video',
    },
    {
      title: 'Jajjaboo',
      url: '/afaanoromoo/popular/read',
    },
  ],
  timezone: 'Africa/Addis_Ababa',
};

export default service;
