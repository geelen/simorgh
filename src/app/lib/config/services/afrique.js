import { C_POSTBOX, C_WHITE } from '@bbc/psammead-styles/colours';
import { latinDiacritics } from '@bbc/gel-foundations/scripts';
import { afrique as brandSVG } from '@bbc/psammead-assets/svgs';
import '@bbc/moment-timezone-include/tz/GMT';
import withContext from '../../../contexts/utils/withContext';
import 'moment/locale/fr';

export const service = {
  default: {
    lang: `fr`,
    articleAuthor: 'https://www.facebook.com/bbcnews',
    articleTimestampPrefix: 'Mise à jour',
    atiAnalyticsAppName: 'news-afrique',
    atiAnalyticsProducerId: '3',
    brandName: 'BBC News Afrique',
    product: 'BBC News',
    serviceLocalizedName: 'Afrique',
    defaultImage: 'https://news.files.bbci.co.uk/ws/img/logos/og/afrique.png',
    defaultImageAltText: 'BBC News Afrique',
    dir: `ltr`,
    externalLinkText: ', lien externe',
    imageCaptionOffscreenText: 'Légende image, ',
    videoCaptionOffscreenText: 'Légende vidéo, ',
    audioCaptionOffscreenText: 'Légende audio, ',
    defaultCaptionOffscreenText: 'Légende, ',
    imageCopyrightOffscreenText: 'Image source, ',
    locale: `fr`,
    datetimeLocale: `fr`,
    service: 'afrique',
    serviceName: 'Afrique',
    themeColor: `${C_POSTBOX}`,
    twitterCreator: '@bbcafrique',
    twitterSite: '@bbcafrique',
    noBylinesPolicy: 'https://www.bbc.com/news/help-41670342#authorexpertise',
    publishingPrinciples: 'https://www.bbc.com/news/help-41670342',
    isTrustProjectParticipant: true,
    script: latinDiacritics,
    manifestPath: '/manifest.json',
    swPath: '/sw.js',
    frontPageTitle: 'Accueil',
    hasRadioSchedule: true,
    theming: {
      brandBackgroundColour: `${C_POSTBOX}`,
      brandLogoColour: `${C_WHITE}`,
    },
    translations: {
      seeAll: 'Tout voir',
      home: 'Accueil',
      currentPage: 'Page en cours',
      skipLinkText: 'Aller au contenu',
      relatedContent: 'Lire plus',
      mediaAssetPage: {
        mediaPlayer: 'Media player',
        audioPlayer: 'Audio player',
        videoPlayer: 'Video player',
      },
      error: {
        404: {
          statusCode: '404',
          title: 'La page ne peut pas être trouvée',
          message:
            "Désolé, nous ne pouvons pas afficher la page que vous cherchez. Essayez s'il vous plaît:",
          solutions: [
            "Double vérification de l'url",
            "Cliquez sur le bouton d'actualisation de votre navigateur",
            "Recherche de cette page à l'aide de la barre de recherche de la BBC",
          ],
          callToActionFirst:
            "Vous pouvez également consulter le page d'accueil du site ",
          callToActionLinkText: 'BBC News Afrique',
          callToActionLast: '',
          callToActionLinkUrl: 'https://www.bbc.com/afrique',
        },
        500: {
          statusCode: '500',
          title: 'Erreur interne du serveur',
          message:
            "Désolé, nous sommes actuellement dans l'impossibilité d'afficher la page que vous recherchez. Essayez s'il vous plaît:",
          solutions: [
            "Cliquez sur le bouton d'actualisation de votre navigateur",
            'Revenir plus tard',
          ],
          callToActionFirst:
            "Vous pouvez également consulter le page d'accueil du site ",
          callToActionLinkText: 'BBC News Afrique',
          callToActionLast: '',
          callToActionLinkUrl: 'https://www.bbc.com/afrique',
        },
      },
      consentBanner: {
        privacy: {
          title:
            'Nous avons mis à jour nos politiques de confidentialité et de cookies',
          description: {
            uk: {
              first:
                "Nous avons apporté d'importants changements à notre politique de confidentialité et de cookies et nous voulons que vous sachiez ce que cela signifie pour vous et vos données.",
              linkText: null,
              last: null,
              linkUrl: null,
            },
            international: {
              first:
                "Nous avons apporté d'importants changements à notre politique de confidentialité et de cookies et nous voulons que vous sachiez ce que cela signifie pour vous et vos données.",
              linkText: null,
              last: null,
              linkUrl: null,
            },
          },
          accept: 'OK',
          reject: 'Découvrez ce qui a changé',
          rejectUrl: 'https://www.bbc.co.uk/usingthebbc/your-data-matters',
        },
        cookie: {
          title: 'Faites-nous savoir que vous acceptez les cookies',
          description: {
            uk: {
              first: 'Nous utilisons ',
              linkText: 'cookies',
              last:
                ' pour vous offrir la meilleur expérience en ligne. Veuillez nous faire savoir si vous acceptez tous ces cookies.',
              linkUrl:
                'https://www.bbc.co.uk/usingthebbc/cookies/what-do-i-need-to-know-about-cookies/',
            },
            international: {
              first:
                'Nous et nos partenaires utilisons des technologies, comme ',
              linkText: 'cookies',
              last:
                ", et collectons des données de navigation pour vous offrir la meilleure expérience en ligne et pour personnaliser le contenu et la publicité qui vous sont présentés. Veuillez nous faire savoir si vous êtes d'accord.",
              linkUrl:
                'https://www.bbc.co.uk/usingthebbc/cookies/what-do-i-need-to-know-about-cookies/',
            },
          },
          accept: "Oui, j'accepte",
          reject: 'Non, dirigez-moi vers les réglages',
          rejectUrl:
            'https://www.bbc.co.uk/usingthebbc/cookies/how-can-i-change-my-bbc-cookie-settings/',
        },
      },
      media: {
        audio: 'Audio',
        photogallery: 'Galerie de photos',
        video: 'Vidéo',
        bbc_afrique_radio: {
          title: 'BBC Afrique Radio',
          subtitle: 'Infos, musique et sports',
        },
        bbc_afrique_tv: {
          title: 'Cash Éco',
          subtitle:
            'Cash éco vous propose chaque jour un résumé de l’actualité économique et financière.',
        },
        listen: 'Listen',
        watch: 'Suivez',
        liveLabel: 'EN DIRECT',
        previousRadioShow: 'Previous radio show',
        nextRadioShow: 'Next radio show',
        duration: 'Duration',
      },
    },
    brandSVG,
    mostRead: {
      header: 'Les plus lus',
      lastUpdated: 'Dernière mise à jour: ',
    },
    footer: {
      trustProjectLink: {
        href: 'https://www.bbc.com/news/help-41670342',
        text: 'Pourquoi vous pouvez faire confiance à BBC News',
      },
      externalLink: {
        href: 'https://www.bbc.co.uk/help/web/links/',
        text: 'Découvrez notre approche en matière de liens externes.',
      },
      links: [
        {
          href: 'https://www.bbc.com/afrique/institutionelles-36826639',
          text: "Conditions d'utilisation",
        },
        {
          href: 'https://www.bbc.com/afrique/institutionelles-36826641',
          text: 'A propos de la BBC',
        },
        {
          href: 'https://www.bbc.com/afrique/institutionelles-36826642',
          text: 'Politique de confidentialité',
        },
        {
          href: 'https://www.bbc.com/usingthebbc/cookies/',
          text: 'Cookies',
        },
        {
          href: 'https://www.bbc.com/afrique/institutionelles-36826644',
          text: 'Contactez la BBC',
        },
      ],
      copyrightText:
        "BBC. La BBC n'est pas responsable du contenu des sites externes.",
    },
    fonts: [],
    timezone: 'GMT',
    navigation: [
      {
        title: 'Accueil',
        url: '/afrique',
      },
      {
        title: 'Afrique',
        url: '/afrique/region',
      },
      {
        title: 'Monde',
        url: '/afrique/monde',
      },
      {
        title: 'Sports',
        url: '/afrique/sports',
      },
      {
        title: 'Economie',
        url: '/afrique/topics/ca170ae3-99c1-48db-9b67-2866f85e7342',
      },
      {
        title: 'Culture',
        url: '/afrique/topics/6a73afa3-ea6b-45c1-80bb-49060b99f864',
      },
      {
        title: 'Au féminin',
        url: '/afrique/38058576',
      },
      {
        title: 'Technologie',
        url: '/afrique/topics/31684f19-84d6-41f6-b033-7ae08098572a',
      },
      {
        title: 'Santé',
        url: '/afrique/topics/c4794229-7f87-43ce-ac0a-6cfcd6d3cef2',
      },
      {
        title: 'Vidéos',
        url: '/afrique/media/video',
      },
      {
        title: 'En images',
        url: '/afrique/media/photogalleries',
      },
      {
        title: 'Nos émissions',
        url: '/afrique/nos_emissions/radio',
      },
    ],
    navigationSection: 'Rubriques',
  },
};

export default withContext(service);
