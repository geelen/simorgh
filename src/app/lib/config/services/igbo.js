import { C_POSTBOX, C_WHITE } from '@bbc/psammead-styles/colours';
import { igbo as brandSVG } from '@bbc/psammead-assets/svgs';
import { cyrillicAndLatin } from '@bbc/gel-foundations/scripts';
import '@bbc/psammead-locales/moment/ig';
import '@bbc/moment-timezone-include/tz/Africa/Lagos';
import withContext from '../../../contexts/utils/withContext';

export const service = {
  default: {
    lang: 'ig',
    articleAuthor: 'https://www.facebook.com/bbcnews',
    articleTimestampPrefix: 'Mgbe ikpeazụ e tinyere ya ozi ọhụrụ',
    atiAnalyticsAppName: 'news-igbo',
    atiAnalyticsProducerId: '53',
    brandName: 'BBC News Ìgbò',
    product: 'BBC News',
    serviceLocalizedName: 'Ìgbò',
    locale: 'ig',
    // valid ISO 639-1 code - this is not the same as lang! see explanation in #3405
    isoLang: 'ig',
    datetimeLocale: 'ig',
    service: 'igbo',
    serviceName: 'Igbo',
    defaultImage: 'https://news.files.bbci.co.uk/ws/img/logos/og/igbo.png',
    defaultImageAltText: 'BBC News Ìgbò',
    dir: 'ltr',
    externalLinkText: ', Site na mpụta',
    imageCaptionOffscreenText: 'Nkọwa foto ',
    videoCaptionOffscreenText: 'Aha onyonyo ',
    audioCaptionOffscreenText: 'Aha nkeananụānụ',
    defaultCaptionOffscreenText: 'Ihe a na-akpọ ya ',
    imageCopyrightOffscreenText: 'Ebe foto si ',
    brandSVG,
    script: cyrillicAndLatin,
    manifestPath: '/manifest.json',
    swPath: '/sw.js',
    frontPageTitle: 'Ogbako',
    fonts: [],
    noBylinesPolicy: 'https://www.bbc.com/news/help-41670342#authorexpertise',
    publishingPrinciples: 'https://www.bbc.com/news/help-41670342',
    isTrustProjectParticipant: true,
    themeColor: `${C_POSTBOX}`,
    twitterCreator: '@BBCNews', // to be updated
    twitterSite: '@BBCNews', // to be updated
    hasRadioSchedule: false,
    theming: {
      brandBackgroundColour: `${C_POSTBOX}`,
      brandLogoColour: `${C_WHITE}`,
    },
    translations: {
      seeAll: 'Lee ha niile',
      home: 'Akụkọ',
      currentPage: 'Peegi ị nọ ugbua',
      skipLinkText: 'Wụga n’ọdịnaya',
      relatedContent: "Ihe ndị ọzọ n'akụkọ a",
      mediaAssetPage: {
        mediaPlayer: 'Media player',
        audioPlayer: 'Audio player',
        videoPlayer: 'Video player',
      },
      error: {
        home: 'Akụkọ',
        currentPage: 'Current page',
        skipLinkText: 'Wụga n’ọdịnaya',
        404: {
          statusCode: '404',
          title: 'Ahụghị ibe akwụkwọ a',
          message:
            'E wela iwe, anyị enweghị ike iwetara gị ibe akwụkwọ ị na-achọ. Nwalie ọzọ:',
          solutions: [
            'Tulee akara URL ahụ ọzọ',
            "Kpatụ ọdụmbị 'Refresh' dị na braụza gị aka",
            'Jiri BBC "Search Bar" chọọ ibe akwụkwọ',
          ],
          callToActionFirst: "Maọbụ, biko gaa n'obi ",
          callToActionLinkText: 'BBC News Ìgbò',
          callToActionLast: '.',
          callToActionLinkUrl: 'https://www.bbc.com/igbo',
        },
        500: {
          statusCode: '500',
          title: 'Mperi',
          message:
            'E wela iwe, anyị enweghị ike iwetara gị ibe akwụkwọ ị na-achọ. Gbalịa ọzọ:',
          solutions: [
            'Kpatụ ọdụmbi Refresh dị na braụza gị aka',
            'Ga alọghachị azụ ma e mechaa',
          ],
          callToActionFirst: "Maọbụ, biko gaa n'obi ",
          callToActionLinkText: 'BBC News Ìgbò',
          callToActionLast: '.',
          callToActionLinkUrl: 'https://www.bbc.com/igbo',
        },
      },
      consentBanner: {
        privacy: {
          title: 'Anyị ewelitela ihe nzuzo anyị nakwa iwu dị na ya.',
          description: {
            uk: {
              first:
                'Anyị agbanwela ụfọdụ ihe dị mkpa nye ihe nzuzo anyị na iwu dị na ya nakwa anyị chọrọ ka ị mata ihe nke a pụtara nye gị na data gị.',
              linkText: null,
              last: null,
              linkUrl: null,
            },
            international: {
              first:
                'Anyị agbanwela ụfọdụ ihe dị mkpa nye ihe nzuzo anyị na iwu dị na ya nakwa anyị chọrọ ka ị mata ihe nke a pụtara nye gị na data gị.',
              linkText: null,
              last: null,
              linkUrl: null,
            },
          },
          accept: 'Ọ dị mma',
          reject: 'Chọpụta ihe gbanwere',
          rejectUrl: 'https://www.bbc.co.uk/usingthebbc/your-data-matters',
        },
        cookie: {
          title: "Ka anyị mara ma i kwenyere n'iwu ndị a",
          description: {
            uk: {
              first: 'Anyị na-eji ',
              linkText: 'cookies',
              last:
                ' enye gị nke kacha mkpa na ịntanetị. Biko me ka anyị mata ma i kwenyere.',
              linkUrl:
                'https://www.bbc.co.uk/usingthebbc/cookies/what-do-i-need-to-know-about-cookies/',
            },
            international: {
              first: 'Anyị na ndị anyị na ha na-emekọrịta na-eji teknụzụdịka ',
              linkText: 'cookies',
              last:
                ', were na-amịị ozi banyere gị iji were na-enye gị ụdị ihe ị chọrọ dịka o si gbasaa ihe akụkọ anyị na ozi azụmahịa anyị na-egosi gị. Biko me ka anyị mata ma ọ dị gị mma.',
              linkUrl:
                'https://www.bbc.co.uk/usingthebbc/cookies/what-do-i-need-to-know-about-cookies/',
            },
          },
          accept: 'Eeh, ekwere m',
          reject: "Mba, duga m n'ebe mwube",
          rejectUrl:
            'https://www.bbc.co.uk/usingthebbc/cookies/how-can-i-change-my-bbc-cookie-settings/',
        },
      },
      media: {
        audio: 'Ọdịyo',
        photogallery: 'Image gallery',
        video: 'Vidio',
        bbc_igbo_radio: {
          title: 'Placeholder title',
          subtitle: 'Placeholder subtitle',
        },
        listen: 'Gee ntị',
        watch: 'Lee',
        liveLabel: 'NA EME UGBU A',
        previousRadioShow: 'Previous radio show',
        nextRadioShow: 'Next radio show',
        duration: 'Duration',
      },
    },
    mostRead: {
      header: 'Akachasị Gụọ',
      lastUpdated: 'Emelitere ikpeazụ na ',
    },
    navigation: [
      {
        title: 'Akụkọ',
        url: '/igbo',
      },
      {
        title: 'Egwuregwu',
        url: '/igbo/topics/4063f80f-cccc-44c8-9449-5ca44e4c8592',
      },
      {
        title: 'Ihe nkiri',
        url: '/igbo/media/video',
      },
      {
        title: 'Nke ka ewuewu',
        url: '/igbo/popular/read',
      },
    ],
    footer: {
      trustProjectLink: {
        href: 'https://www.bbc.com/news/help-41670342',
        text: 'Ihe mere ị ga-eji nwee ntụkwasiobi na BBC News',
      },
      externalLink: {
        href: 'https://www.bbc.co.uk/help/web/links/',
        text: 'Usoro anyị maka njikọ dị na mpụga.',
      },
      links: [
        {
          href: 'https://www.bbc.co.uk/usingthebbc/terms',
          text: 'Usoro Ojiji',
        },
        {
          href: 'https://www.bbc.co.uk/aboutthebbc',
          text: 'Gbasara BBC',
        },
        {
          href: 'https://www.bbc.co.uk/usingthebbc/privacy/',
          text: 'Iwu Nzuzo',
        },
        {
          href: 'https://www.bbc.co.uk/usingthebbc/cookies/',
          text: 'Kuki',
        },
        {
          href: 'https://www.bbc.com/igbo/institutional-43090448',
          text: 'Kpọtụrụ BBC',
        },
      ],
      navigationSection: 'Ngalaba',
      copyrightText:
        'BBC. BBC anaghị ahụta maka ọdịnaya nke saịtị ndị dị na mpụga.',
    },
    timezone: 'Africa/Lagos',
  },
};

export default withContext(service);
