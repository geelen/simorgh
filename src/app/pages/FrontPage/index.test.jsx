import React from 'react';
import { render } from '@testing-library/react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import FrontPageContainer from './index';
import { service as igboConfig } from '#lib/config/services/igbo';
import igboData from '#data/igbo/frontpage';
import toggleReducer from '../../reducers/ToggleReducer';
import defaultToggles from '#lib/config/toggles';

// explicitly ignore console.log errors for Article/index:getInitialProps() error logging
global.console.log = jest.fn();

const defaultProps = {
  isAmp: false,
  pageType: 'frontPage',
  service: 'news',
  pathname: '/pathname',
  data: { status: 200 },
};

jest.mock('../../containers/PageHandlers/withVariant', () => Component => {
  const VariantContainer = props => (
    <div id="VariantContainer">
      <Component {...props} />
    </div>
  );

  return VariantContainer;
});

jest.mock('../../containers/PageHandlers/withContexts', () => Component => {
  const DataContainer = props => (
    <div id="ContextsContainer">
      <Component {...props} />
    </div>
  );

  return DataContainer;
});

jest.mock('../../containers/PageHandlers/withPageWrapper', () => Component => {
  const PageWrapperContainer = props => (
    <div id="PageWrapperContainer">
      <Component {...props} />
    </div>
  );

  return PageWrapperContainer;
});

jest.mock('../../containers/PageHandlers/withLoading', () => Component => {
  const LoadingContainer = props => (
    <div id="LoadingContainer">
      <Component {...props} />
    </div>
  );

  return LoadingContainer;
});

jest.mock('../../containers/PageHandlers/withError', () => Component => {
  const ErrorContainer = props => (
    <div id="ErrorContainer">
      <Component {...props} />
    </div>
  );

  return ErrorContainer;
});

jest.mock('../../containers/PageHandlers/withData', () => Component => {
  const DataContainer = props => (
    <div id="DataContainer">
      <Component {...props} />
    </div>
  );

  return DataContainer;
});

jest.mock('../../containers/PageHandlers/withContexts', () => Component => {
  const ContextsContainer = props => (
    <div id="ContextsContainer">
      <Component {...props} />
    </div>
  );

  return ContextsContainer;
});

jest.mock('../../containers/FrontPageMain', () => {
  return jest.fn().mockReturnValue(<div>FrontPageMain</div>);
});

describe('FrontPageContainer', () => {
  describe('Component', () => {
    describe('Composing the Front Page Container using the page handlers', () => {
      shouldMatchSnapshot(
        'should compose frontPageContainer with the Page Handler in the correct order',
        <FrontPageContainer {...defaultProps} />,
      );
    });

    describe('Assertions', () => {
      let FrontPageComponent;
      beforeAll(() => {
        jest.resetModules();
        jest.unmock('../../containers/PageHandlers/withError');
        jest.unmock('../../containers/PageHandlers/withLoading');
        jest.unmock('../../containers/PageHandlers/withData');

        jest.mock('react', () => {
          const original = jest.requireActual('react');
          return {
            ...original,
            useContext: jest.fn(),
            useReducer: jest.fn(),
            useState: jest.fn(),
          };
        });

        const { useContext, useReducer, useState } = jest.requireMock('react');
        useContext.mockReturnValue(igboConfig.default);
        FrontPageComponent = jest.requireActual('.').default;
        useReducer.mockReturnValue([toggleReducer, defaultToggles]);
        useState.mockImplementation(input => [input, () => {}]);
      });

      it('should not render frontpage if still loading', () => {
        const { container } = render(
          <FrontPageComponent {...defaultProps} loading />,
        );
        const { textContent } = container.querySelector('main');

        expect(textContent).toEqual('');
        expect(textContent).not.toContain('FrontPageMain');
      });

      it('should render error page when an error occurs', () => {
        const { container } = render(
          <FrontPageComponent {...defaultProps} error={new Error('oh no')} />,
        );

        const { textContent } = container.querySelector('main');
        expect(textContent).toContain('Mperi');
        expect(textContent).not.toContain('FrontPageMain');
      });

      it('should render the frontpage with data', () => {
        const pageData = igboData;
        const status = 200;

        const frontPageMainMock = jest.requireMock(
          '../../containers/FrontPageMain',
        );
        const { container } = render(
          <FrontPageComponent
            {...defaultProps}
            error={null}
            service="igbo"
            pageData={pageData}
            status={status}
          />,
        );

        expect(frontPageMainMock.mock.calls).toHaveLength(1);
        expect(frontPageMainMock.mock.calls[0][0]).toEqual({
          frontPageData: igboData,
        });
        expect(container.textContent).toEqual('FrontPageMain');
      });
    });
  });
});
