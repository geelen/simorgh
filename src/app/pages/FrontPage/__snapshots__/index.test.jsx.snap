// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrontPageContainer Component Composing the Front Page Container using the page handlers should compose frontPageContainer with the Page Handler in the correct order 1`] = `
<div
  id="VariantContainer"
>
  <div
    id="ContextsContainer"
  >
    <div
      id="PageWrapperContainer"
    >
      <div
        id="LoadingContainer"
      >
        <div
          id="ErrorContainer"
        >
          <div
            id="DataContainer"
          >
            <div>
              FrontPageMain
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
